### PR TITLE
Deliverable 1

### PR DESCRIPTION
**Pull request by omicto**
# Mini-Challenge 1: Core Concepts and Styling

## Before you start:
1. Read carefully [the instructions for the project](https://gist.github.com/erickwize/d7311bfc972080c162c43cbb7dc80587), please make sure that you clearly understand all the requirements.
2. Fork [this repo](https://github.com/erickwize/react-certification-2020). Please make sure that your new repo is publicly accesible.

> **NOTE:* You should use the provided codebase only as a guide for structuring your application. Feel free to add, remove, or change anything if you consider it necessary.*

## The challenge:
1. Create the UI layout for your app, it should include the following elements:
  - 1.1 Header: Similar to the one in our [demo app](https://react-certification-2020.netlify.app/). For now, the search field, theme toggle, login button, and other navigation links should be disabled, i.e., you only need to create the corresponding UI elements without the functionality.
  - 1.2 Content: The content view corresponding to the section that will be renderered (in this case, the Home View).
2. Create the **Home View** and display the list of items from [this mock file](https://gist.github.com/jparciga/1d4dd34fb06ba74237f8966e2e777ff5).
3. Apply css styles to your components using the [styled-components](https://styled-components.com/) approach.

## Acceptance Criteria
- [ ] The header is rendered correctly.
- [ ] A list of videos obtained from a mock file is displayed in the **Home View**.
- [ ] CSS styles are applied correctly using the styled-components approach.

## Bonus Points
- [ ] No warnings or errors are logged in the browser console.
- [ ] The UI is responsive

----
# Mini-Challenge 2: Intro to Testing

## Answer the following questions first
1. Using `create-react-app`, what do we need to set up for testing?
2. What components are worth to test in your development?
3. Can you apply TDD once you already created components?

## The Challenge

> **NOTE:* Apply TDD as much as you can.*

1. Find any sub-routine, extract that logic into a separate file and test it out.
2. Run test coverage and save results into a .txt file at ./ - root level. Can you make the total coverage to be above of 60%?

## Acceptance Criteria
- [ ] Meaningful test cases were implemented for components and sub-routines logic.
- [ ] All the test cases were successful.

## Bonus Points
- [ ] Test coverage is above 60%.

### CODE REVIEW

**Comment by omicto** [comment]
Answers for the 2nd mini challenge are in the root of the project, but I'm adding them here just in case:

# Mini Challenge 2 answers

> Using `create-react-app`, what do we need to set up for testing?

Apps created with _CRA_ come with Jest preconfigured so we can run tests right away. However we can install `react-testing-library` to test components in a way that is more similar to how a user will interact with our app.

> What components are worth to test in your development?

We should strive to test all of our components. A comprehensive suite of tests allows a faster feedback loop when making changes in the code, given that all of our components are subject to change in the future, it may be worth to write tests for all of them.

> Can you apply TDD once you already created components?

Yes, the first step would be to create the missing tests for the existing components (and maybe refactor them, given that they probably were not created with tests in mind).

From that point on, you can now start by writing the tests for whatever new components or functionality you need to add.

File: src/components/Navigation/Navigation.component.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    color: white;
+  }
+`;
+
+const Navigation = () => {
+  return (
+    <NavBar>
+      <Link to="/">WizeTube</Link>
+      <SearchBar initialValue="wizeline" />
+      <ToggleButton disabled>Dark mode</ToggleButton>
```
Comment (MartinMolinero) [inline src/components/Navigation/Navigation.component.jsx:33]: Nice use of the atomic design principles we saw on the workshop 👏

**Review by MartinMolinero — APPROVED** [review]
You are going in a good direction with your project, making good progress. Try increasing your testing coverage, by testing the components and functions you created.

File: src/components/Navigation/Navigation.component.jsx (hunk 1 above)
Comment (omicto) [reply src/components/Navigation/Navigation.component.jsx:33]: Thanks for the comments, will work on increasing the coverage 💪

**Pull request merged**